plugin/evm/vm.go

```diff
@@ -325,17 +325,13 @@ func (vm *VM) Initialize(
 	}
 
 	// Handle custom fee recipient
-	vm.ethConfig.Miner.Etherbase = constants.BlackholeAddr
 	if common.IsHexAddress(vm.config.FeeRecipient) {
-		if g.Config.AllowFeeRecipients {
-			address := common.HexToAddress(vm.config.FeeRecipient)
-			log.Info("Setting fee recipient", "address", address)
-			vm.ethConfig.Miner.Etherbase = address
-		} else {
-			return errors.New("cannot specify a custom fee recipient on this blockchain")
-		}
-	} else if g.Config.AllowFeeRecipients {
-		log.Warn("Chain enabled `AllowFeeRecipients`, but chain config has not specified any coinbase address. Defaulting to the blackhole address.")
+		address := common.HexToAddress(vm.config.FeeRecipient)
+		log.Info("Setting fee recipient", "address", address)
+		vm.ethConfig.Miner.Etherbase = address
+	} else {
+		log.Warn("Config has not specified any coinbase address. Defaulting to the blackhole address.")
```

Comment:
It will still be handled in miner/worker while setting header coinbase. Before, it's guaranteed to know if this was activated in genesis, so we were able to show a warning message or set etherbase to blackhole address accordingly. i.e if allow fee, show warning if no address is specified; otherwise set etherbase it to blackhole address. However with that precompile a network can always switch to allow fee recipients after the genesis. In that case we cannot forcefully set the etherbase address to blackhole address otherwise they would always send fees to blackhole address if allow fee is enabled in the future. so we have to keep etherbase as it is. instead of setting the etherbase here we set the coinbase in the miner/worker just before committing the header.